lib/src/rocket.rs

```diff
@@ -214,20 +243,24 @@ impl Rocket {
                 response
             }
             Outcome::Forward(data) => {
-                // Rust thinks `request` is still borrowed here, but it's
-                // obviously not (data has nothing to do with it), so we
-                // convince it to give us another mutable reference.
-                // TODO: Use something that is well defined, like UnsafeCell.
-                // But that causes variance issues...so wait for NLL.
-                let request: &'r mut Request<'s> =
-                    unsafe { (&mut *(request as *const _ as *mut _)) };
 
                 // There was no matching route. Autohandle `HEAD` requests.
                 if request.method() == Method::Head {
                     info_!("Autohandling {} request.", Paint::white("HEAD"));
+
+                    // Rust thinks `request` is still borrowed here, but it's
+                    // obviously not (data has nothing to do with it), so we
+                    // convince it to give us another mutable reference.
+                    // TODO: Use something that is well defined, like UnsafeCell.
```

Comment:
What is the status of this TODO -- is there a better solution than the `unsafe` block? NLL is available now.

Reply:
I'm not sure if NLL resolves this particular issue, but I'd be curious to know if it does. That being said, there are far too many soundness problems with the NLL implementation as it stands for me to be comfortable enabling it in Rocket.

Reply:
I tried to use it with NLL, but it does not solve the problem. Do you have any suggestions?

Reply:
You can leave this as is.